clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -561,6 +584,91 @@ scheduler_tmpl_t<RecordType, ReaderType>::init(
     return STATUS_SUCCESS;
 }
 
+template <typename RecordType, typename ReaderType>
+std::string
+scheduler_tmpl_t<RecordType, ReaderType>::recorded_schedule_component_name(
+    output_ordinal_t output)
+{
+    static const char *const SCHED_CHUNK_PREFIX = "output.";
+    std::ostringstream name;
+    name << SCHED_CHUNK_PREFIX << std::setfill('0') << std::setw(4) << output;
+    return name.str();
+}
+
+template <typename RecordType, typename ReaderType>
+typename scheduler_tmpl_t<RecordType, ReaderType>::scheduler_status_t
+scheduler_tmpl_t<RecordType, ReaderType>::write_recorded_schedule()
+{
+    if (options_.record_ostream == nullptr)
+        return STATUS_ERROR_INVALID_PARAMETER;
+    std::lock_guard<std::mutex> guard(sched_lock_);
+    for (int i = 0; i < static_cast<int>(outputs_.size()); i++) {
```

Comment:
nit: use ++i consistently throughout

Reply:
It looks like this file is consistently using i++ so it is already consistent?

Reply:
Some other files in the PR were using ++i, the tests I think. ++i is generally more efficient, so we can update all instances in this file, optionally.

Reply:
For an int it doesn't matter and i++ is the traditional C style -- but I do see the Google C++ style guide votes for ++i.  Sure, changed.